tests/ttnn/unit_tests/operations/test_moreh_softmin.py

```diff
@@ -397,3 +389,97 @@ def test_softmin_backward_optional_output_tensor(shape_dim, optional_output_tens
+    torch.manual_seed(0)
+
+    for i in range(2):
+        x = torch.randint(low=0, high=4, size=shape).to(torch.bfloat16) + i
+
+        dev_x = ttnn.Tensor(x, ttnn.bfloat16).to(ttnn.TILE_LAYOUT).to(device)
+
+        tt_cpu = F.softmin(x, dim)
+        tt_npu = ttnn.operations.moreh.softmin(dev_x, dim, strategy=strategy)
+        if i == 0:
+            num_program_cache_entries = device.num_program_cache_entries()
+            assert num_program_cache_entries > 0
+        else:
+            assert device.num_program_cache_entries() == num_program_cache_entries
+        torch_dummy = torch.randn([32, 32])
+        tt_dummy = to_npu(torch_dummy, device)
+
+    assert list(tt_npu.shape.with_tile_padding()) == list(tt_cpu.shape)
+    tt_dev = tt_npu.cpu().to(ttnn.ROW_MAJOR_LAYOUT).to_torch().to(torch.bfloat16)
+
+    rtol = atol = 0.05
+    passing, out = comp_allclose_and_pcc(tt_cpu, tt_dev, rtol=rtol, atol=atol)
+    logger.debug(out)
+    assert passing
+
+
+def softmin_backward_step(dev_y, dev_dy, dim, strategy, device, num_program_cache_entries=None):
+    """
+    Runs a single step of softmin_backward and checks the program cache if needed.
+    """
+    tt_npu = ttnn.operations.moreh.softmin_backward(dev_y, dev_dy, dim, strategy=strategy)
+
+    if num_program_cache_entries is not None:
+        assert device.num_program_cache_entries() == num_program_cache_entries
+    else:
+        num_program_cache_entries = device.num_program_cache_entries()
+        assert num_program_cache_entries > 0
+
+    torch_dummy = torch.randn([32, 32])
+    tt_dummy = to_npu(torch_dummy, device)
+
+    return tt_npu, num_program_cache_entries
+
+
+@pytest.mark.parametrize(
+    "shape_dim_strategy",
+    [
+        [[32, 32], 1, ttnn.operations.moreh.SoftmaxBackwardOpParallelizationStrategy.SMALL_W],
+        [[32, 32], 0, ttnn.operations.moreh.SoftmaxBackwardOpParallelizationStrategy.SMALL_H],
+        [[2, 3, 32 * 4, 32 * 5], 3, ttnn.operations.moreh.SoftmaxBackwardOpParallelizationStrategy.LARGE_W],
+        [[2, 3, 32 * 4, 32 * 5], 2, ttnn.operations.moreh.SoftmaxBackwardOpParallelizationStrategy.LARGE_H],
+        [[1, 15, 32, 32], 1, ttnn.operations.moreh.SoftmaxBackwardOpParallelizationStrategy.LARGE_C],
+    ],
+)
+def test_softmin_backward_callback(shape_dim_strategy, device, use_program_cache):
+    shape, dim, strategy = shape_dim_strategy
+    torch.manual_seed(0)
+
+    x = torch.randint(low=0, high=4, size=shape).to(torch.bfloat16).requires_grad_(True)
+    y = F.softmin(x, dim)
+    dev_y = ttnn.Tensor(y, ttnn.bfloat16).to(ttnn.TILE_LAYOUT).to(device)
+
+    dy = torch.randint(low=0, high=4, size=shape).to(torch.bfloat16)
+    dev_dy = ttnn.Tensor(dy, ttnn.bfloat16).to(ttnn.TILE_LAYOUT).to(device)
+
+    y.backward(dy)
+
+    num_program_cache_entries = None
+    for i in range(2):
+        tt_npu, num_program_cache_entries = softmin_backward_step(
+            dev_y, dev_dy, dim, strategy, device, num_program_cache_entries
+        )
```

Comment:
This has the same issue as the forward. The input values of `y` and `dy` in the backward loop are the same in both iterations.

Reply:
`for i in range(2):
        x = torch.randint(low=0, high=4, size=shape).to(torch.bfloat16).requires_grad_(True)
        y = F.log_softmax(x, dim)
        dev_y = ttnn.Tensor(y, ttnn.bfloat16).to(ttnn.TILE_LAYOUT).to(device)

        dy = torch.randint(low=0, high=4, size=shape).to(torch.bfloat16)
        dev_dy = ttnn.Tensor(dy, ttnn.bfloat16).to(ttnn.TILE_LAYOUT).to(device)

        y.backward(dy)

        tt_npu, num_program_cache_entries = logsoftmax_backward_step(
            dev_y, dev_dy, dim, strategy, device, num_program_cache_entries
        )
`
This should be okay right?

Reply:
Yes, that sounds good.